### PR TITLE
feat(curriculum): Add progress indicators to maps

### LINKED ISSUE #50412: Add progress indicators to maps
Here's some mocks:

### PR DESCRIPTION
**Pull request by CallmeHongmaybe**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #50412

### CODE REVIEW

File: config/superblocks.ts
Comment (naomi-lgbt) [inline config/superblocks.ts:74]: This shouldn't be changed. The previous responsive web design is still considered legacy?

**Comment by bbsmooth** [comment]
Just a few quick observations for now:

- Since these are ordered lists, I think we should use `ol` instead of `ul` for them.
- You can't have `h2` elements as direct children of list elements. Each of these stages will probably need to be it's own list with its corresponding `h2` right above it.

Also, all of the numbers have the black ribbon background. Are we still going to distinguish completed vs. uncompleted by giving the uncompleted numbers a different look?

File: config/superblocks.ts
Comment (CallmeHongmaybe) [reply config/superblocks.ts:74]: I guess so. 
I arranged the superblocks by following [the design](https://github.com/freeCodeCamp/freeCodeCamp/issues/50412#issuecomment-1585086022) here

**Comment by CallmeHongmaybe** [comment]
@bbsmooth regarding your second bullet point, do you think the elements should be ordered like this ?
```
<>
     <Spacer size='small' />
        <h2>
            Stage {indx + 1}: {stage}
        </h2>
        <ol>
           {blockElement}
        </ol>
</>
```
About the ribbon thing, since the certified user have all the certifications there is, it's expected that all numbers are included with the ribbon. 
I tried using MongoDB Compass to change the user certifications list, but I couldn't open it for who-knows reason. I could use shell to update, but I thought I will hear you guys out about this first.

**Review by huyenltnguyen — COMMENTED** [review]
I have some suggestion on the implementation of the stage section, but I'd prefer not overwhelming you. Let's address the current feedback first 🙂 

---
There is an alignment issue with the icon, and... we will need the icon to work with dark theme 😅 

Screenshot 2023-08-07 at 12 39 19

File: client/src/components/Map/index.tsx
Comment (CallmeHongmaybe) [reply client/src/components/Map/index.tsx:45]: Not anymore.

File: client/src/components/Map/index.tsx
Comment (CallmeHongmaybe) [reply client/src/components/Map/index.tsx:75]: So far I haven't got any type errors. Just being nitpicky about types sometimes. Thanks for pointing out.

File: client/src/components/Map/index.tsx
Comment (CallmeHongmaybe) [reply client/src/components/Map/index.tsx:133]: You could, but you'd still need to declare its types somewhere.

File: client/src/components/Map/index.tsx
Comment (CallmeHongmaybe) [reply client/src/components/Map/index.tsx:140]: I don't think `MapState` would be a suitable name. Let's call it `MapList` shall we ?

File: config/superblocks.ts
Comment (ojeytonwilliams) [inline config/superblocks.ts:41]: This should never have had string values. I've created https://github.com/freeCodeCamp/freeCodeCamp/pull/51178 to fix that, but please revert this change. It should be fine to use the enums directly.

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]

File: config/superblocks.ts
Comment (CallmeHongmaybe) [reply config/superblocks.ts:41]: Alrighty.

**Review by huyenltnguyen — COMMENTED** [review]
@CallmeHongmaybe Thank you for the response! Please help keep the comments unresolved until there are code changes and the reviewers have a chance to look at your changes/responses 🙂

**Comment by CallmeHongmaybe** [comment]
@huyenltnguyen sure. The lint requires me to state the type of the two variables above, so I decided to declare the type of the variables in one line instead.

```js
(certSlugTypeMap as { [key: string]: string })[cert.certSlug] === (superBlockCertTypeMap as { [key: string]: string })[superBlock]
                        ```

File: client/src/assets/icons/completion-ribbon.tsx
Comment (CallmeHongmaybe) [reply client/src/assets/icons/completion-ribbon.tsx:10]: Regarding the `sr-only` texts, can you give me an example on how it's done ? If I use this classname on the block title, the text won't appear at all.

File: client/src/components/Map/index.tsx
Comment (bbsmooth) [inline client/src/components/Map/index.tsx:139]:
```suggestion
        <ul>
```

This should be an `ol` since order is important in these lists.

File: client/src/components/Map/index.tsx
Comment (bbsmooth) [inline client/src/components/Map/index.tsx:146]: 'h2` elements cannot be direct descendants of a list. Each stage should be it's own list and the `h2` for the stage should come directly before the list.

File: client/src/assets/icons/completion-ribbon.tsx
Comment (Sboonny) [inline client/src/assets/icons/completion-ribbon.tsx:20]: This approach is unneeded, because the component isn't an external one that require a specific way to change its style.

Please change the style through CSS.

File: client/src/assets/icons/completion-ribbon.tsx
Comment (Sboonny) [inline client/src/assets/icons/completion-ribbon.tsx:9]: This part is unneeded, if you are changing it through CSS.

**Review by Sboonny — CHANGES_REQUESTED** [review]

**Comment by CallmeHongmaybe** [comment]
When I investigated the failed Cypress tests, my thoughts about them as follows:

In ["Should render a curriculum map"] 
(https://github.com/freeCodeCamp/freeCodeCamp/actions/runs/5807008783/job/15741440717?pr=51127#step:13:380) test case, the query selector needs to be updated. 

I have no idea how to fix the Lint error inside the [link](https://github.com/freeCodeCamp/freeCodeCamp/actions/runs/5807008775/job/15741047659?pr=51127)

**Comment by Sboonny** [comment]
running `pnpm format:prettier` should format that file. 

if it doesn't, visit the file in your editor and format it there with prettier.

**Comment by CallmeHongmaybe** [comment]
Thanks. Should I do anything about the test case aside from running the lint ?

**Comment by naomi-lgbt** [comment]
The conflicts will need to be resolved. Let us know if you need assistance with that.

**Comment by CallmeHongmaybe** [comment]
Kindly check again since the conflict is resolved.

**Comment by naomi-lgbt** [comment]
It does look like the cypress tests will need to be updated.

**Comment by moT01** [comment]
Sorry, need to temporarily block this for the next week while we work on add another superblock and make some slight changes to the map. Feel free to keep working on it, though.

**Comment by CallmeHongmaybe** [comment]
> In [Should render a curriculum map](https://github.com/freeCodeCamp/freeCodeCamp/actions/runs/5807008783/job/15741440717?pr=51127#step:13:380) test case, the query selector needs to be updated.

Sure. I'll work on changing the query selector as mentioned before.

**Comment by moT01** [comment]
Unblocked. We made the C# live, this will need a rebase and possibly some adjustments on the design.

**Comment by ahmaxed** [comment]
Tom's suggestion sounds good to me. @CallmeHongmaybe, please let us know if you need help with completing this pr.

**Comment by CallmeHongmaybe** [comment]
@ahmaxed I'm on it. Sorry for the long wait.

**Comment by Sembauke** [comment]
Closing this in favour of #52686 , feel free to go ahead and work on this or leave it to the other contributor.

**Pull request closed without merging**
